app/controllers/request-to-go-live/organisation-address/post.controller.js

```diff
@@ -82,19 +89,42 @@ module.exports = (req, res) => {
   const errors = validate(formFields)
 
   if (lodash.isEmpty(errors)) {
-    // TODO: handle submission
-    renderErrorView(req, res)
+    const updateRequest = new ServiceUpdateRequest()
+      .replace(validPaths.merchantDetails.addressLine1, formFields[clientFormIds.addressLine1])
+      .replace(validPaths.merchantDetails.addressLine2, formFields[clientFormIds.addressLine2])
+      .replace(validPaths.merchantDetails.addressCity, formFields[clientFormIds.addressCity])
+      .replace(validPaths.merchantDetails.addressPostcode, formFields[clientFormIds.addressPostcode])
+      .replace(validPaths.merchantDetails.addressCountry, formFields[clientFormIds.addressCountry])
+      .replace(validPaths.merchantDetails.telephoneNumber, formFields[clientFormIds.telephoneNumber])
+      .replace(validPaths.currentGoLiveStage, goLiveStage.ENTERED_ORGANISATION_ADDRESS)
+      .formatPayload()
+
+    const updatedService = await updateService(req.service.externalId, updateRequest, req.correlationId)
+
+    res.redirect(
+      303,
+      goLiveStageToNextPagePath[updatedService.currentGoLiveStage].replace(':externalServiceId', req.service.externalId)
+    )
   } else {
     lodash.set(req, 'session.pageData.requestToGoLive.organisationAddress', {
       success: false,
       errors: errors,
-      address_line1: formFields[ADDRESS_LINE1_FIELD],
-      address_line2: formFields[ADDRESS_LINE2_FIELD],
-      address_city: formFields[ADDRESS_CITY_FIELD],
-      address_postcode: formFields[ADDRESS_POSTCODE_FIELD],
-      address_country: formFields[ADDRESS_COUNTRY_FIELD],
-      telephone_number: formFields[TELEPHONE_NUMBER_FIELD]
+      address_line1: formFields[clientFormIds.addressLine1],
+      address_line2: formFields[clientFormIds.addressLine2],
+      address_city: formFields[clientFormIds.addressCity],
+      address_postcode: formFields[clientFormIds.addressPostcode],
+      address_country: formFields[clientFormIds.addressCountry],
+      telephone_number: formFields[clientFormIds.telephoneNumber]
     })
-    return res.redirect(303, requestToGoLive.organisationAddress.replace(':externalServiceId', req.service.externalId))
+    res.redirect(303, requestToGoLive.organisationAddress.replace(':externalServiceId', req.service.externalId))
+  }
+}
+
+module.exports = async function (req, res) {
+  try {
+    await submit(req, res)
```

Comment:
I know this is a pattern we have in a number of places but I'm generally uncomfortable with passing the entire `req` and `res` object around.

I don't have the full context for this feature but could this be split into something like: 
```js 
try { 
  const { errors, form } = validateFormBody(req.body) 
  
  if (lodash.isEmpty(errors) { 
    // use form values - build service update request and `await` submit
  } else { 
    // use error values - render error view
  }
} catch (error) { 
  logger.error(`Error submitting organisation address - ${err.stack}`)
  renderErrorView(req, res)
}
```

Reply:
nice suggestion, i'll do it like this